src/gate_detect/src/Gate.cpp

```diff
@@ -0,0 +1,189 @@
+#include "Gate.h"
+
+Gate::Gate() {
+    lowVertThresh = 50; // Sets the maximum deviation from start to finish that
+                        // will constitute a vertical line
+    lowHorThresh = 100; // Sets the maximum deviation from start to finish that
+                        // will constitute a horizontal line
+    cannyLow = 50; // Sets the low threshold for the canny filter, adjust if dst
+                   // is not displaying all lines
+    cannyHigh = 200; // Sets the high threshold for the canny filter, adjust if
+                     // dst not displaying all lines
+    poleMax =
+    200; // Sets the minimum threshold for two lines to be called a pole
+    m                = -0.002673; // Readjust when calibrated
+    b                = 5.134;     // Distance from gate = m * x + b
+    camera_res_width = 1920;
+}
+
+std::vector<float> Gate::initialize(const cv::Mat matin) {
+    cv::Canny(matin, dst, cannyLow, cannyHigh, 3);
+
+    cv::HoughLinesP(dst, lines, 1, CV_PI / 180, 60, 50, 50);
+
+    if (lines.empty()) return std::vector<float>(9, 0);
+
+    _vertLines = filterVertLines(lines);
```

Comment:
can any of the variables used here be temporary variables rather than members?

Reply:
Made lots of vars temporary